tidy3d/web/webapi.py

```diff
@@ -436,6 +467,25 @@ def get_tasks(
     return out_dict
 
 
+def estimate_cost(task_id: str) -> float:
+    """Compute the maximum flex unit charge for a given task, assuming the simulation runs for
+    the full ``run_time``. If early shut-off is triggered, the cost is adjusted proporionately.
+    """
+
+    method = f"tidy3d/tasks/{task_id}/metadata"
+    data = {}
+
+    # do not pass protocol version if mapping is missing or needs an override.
+    if DEFAULT_CONFIG.solver_version:
+        data["solverVersion"] = DEFAULT_CONFIG.solver_version
+    else:
+        data["protocolVersion"] = __version__
+
+    resp = http.post(method, data=data)
```

Comment:
might want to wrap this in try except too:

```python
    try:
        resp = http.post(method=method, data=data)
    except Exception as e:
        log.warning("could not get flex unit information.")
    return resp.get("flex_unit")
```

Does this fit with the expected use?

Reply:
All http calls are already wrapped in a `handle_response` decorator in `httputils.py`.

Reply:
That said actually I think you're right, we may want to wrap this so that it doesn't error out if it doesn't get the cost for some reason. Technically if that happens chances are the simulation won't be able to run correctly anyway (something's wrong with the web settings?) But it may be useful for our develop purposes.

Reply:
yea.. well it could just be something wrong with our estimator, or the flex unit was not included in the response.

Reply:
If it wasn't included in the response, it will just print None. Anyway the reason why I'm saying this should never happen is because the same metadata service is run when a job is started, and is required for the job to be picked up by the solver.